File: src/main/java/com/heybys/oddments/domain/order/OrderId.java
package com.heybys.oddments.domain.order;

import com.heybys.oddments.base.domain.LongTypeIdentifier;
import com.heybys.oddments.base.jpa.hibernate.LongTypeIdentifierJavaType;

public class OrderId extends LongTypeIdentifier {

    public OrderId(Long id) {
        super(id);
    }

    public static class OrderIdJavaType extends LongTypeIdentifierJavaType<OrderId> {
        public OrderIdJavaType() {
            super(OrderId.class);
        }
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/domain/order/OrderId.kt
package com.heybys.oddments.fooddelivery.domain.order

import com.heybys.oddments.base.domain.LongTypeIdentifier
import com.heybys.oddments.base.jpa.hibernate.LongTypeIdentifierJavaType

class OrderId(id: Long) : LongTypeIdentifier(id) {

    class OrderIdJavaType : LongTypeIdentifierJavaType<OrderId>(OrderId::class.java)
}
